File: erlang/pro1.erl
-module(pro1).
-export([pro1/1]).

rec(N, Sum, Num) ->
    if
	N >= Num -> Sum;
	0 == N rem 3 -> rec(N+1, Sum+N, Num);
	0 == N rem 5 -> rec(N+1, Sum+N, Num);
	true -> rec(N+1, Sum, Num)
    end.

pro1(Num) ->
    rec(1, 0, Num).

File: rust/pro1.rs
fn main() {
    let answer = pro1(1000);
    println!("答案为{}", answer);
}

fn pro1(num: i32) -> i32 {
    let mut sum = 0;
    for i in 1..num {
        if i % 3 == 0 || i % 5 == 0 {
            sum += i;
        }
    }
    sum
}
